vindex/cmd/sumdbverify/client.go

```diff
@@ -0,0 +1,260 @@
+	if err != nil {
+		return fmt.Errorf("failed to parse go.mod file: %v", err)
+	}
+	modName := modPath.Module.Mod.Path
+
+	eg, egctx := errgroup.WithContext(ctx)
+	var versions map[string]modData
+	var tags map[string]struct{}
+
+	eg.Go(func() error {
+		// This function gets all version info from the verifiable index.
+		vic := newVIndexClientFromFlags()
+
+		versions, err = queryIndex(egctx, vic, modName)
+		if err != nil {
+			return fmt.Errorf("error querying index: %v", err)
+		}
+		return nil
+	})
+	eg.Go(func() error {
+		// This function gets all tags from the local git checkout.
+		rawTags, err := queryTags(egctx, *modRoot)
+		if err != nil {
+			return fmt.Errorf("error enumerating git tags: %v", err)
+		}
+		tags = make(map[string]struct{}, len(rawTags))
+		for _, t := range rawTags {
+			tags[t] = struct{}{}
+		}
+		return nil
+	})
+	if err := eg.Wait(); err != nil {
+		return err
+	}
+
+	// Create a sorted slice of versions
+	sv := make([]string, 0, len(versions))
+	for v := range versions {
+		sv = append(sv, v)
+	}
+	semver.Sort(sv)
+
+	fmt.Println(modName)
+	for _, v := range sv {
+		d := versions[v]
+		presence := "✅ found in git tags"
+		if _, found := tags[v]; !found {
+			presence = "❌ missing from git tags"
+		}
+
+		fmt.Printf("%s found at index %d: %s\n", v, d.index, presence)
+		delete(tags, v)
+	}
+
+	if len(tags) > 0 {
+		fmt.Println("----------")
+		fmt.Println("> INFO: The tagged versions below were never downloaded via the Module Proxy")
+		for t := range tags {
+			fmt.Printf("%s found locally but missing from SumDB\n", t)
+		}
+	}
+	return nil
+}
+
+type modData struct {
+	index   uint64
+	zipHash []byte
+	modHash []byte
+}
+
+func queryTags(ctx context.Context, modRoot string) ([]string, error) {
+	cmd := exec.CommandContext(ctx, "git", "tag")
+	cmd.Dir = modRoot
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, fmt.Errorf("failed to run git tag: %w", err)
+	}
+	trimmed := strings.TrimSpace(string(out))
+	if trimmed == "" {
+		return []string{}, nil
+	}
+	tags := strings.Split(trimmed, "\n")
+	return tags, nil
+}
+
+func queryIndex(ctx context.Context, vic *client.VIndexClient, modName string) (map[string]modData, error) {
+	idxes, inCp, err := vic.Lookup(ctx, modName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to look up key: %v", err)
+	}
+
+	versions := make(map[string]modData, len(idxes))
+
+	lr := newInputLogClientFromFlags()
+	klog.V(1).Infof("Dereferencing %d pointers", len(idxes))
+	for leaf, err := range lr.Dereference(ctx, inCp, idxes) {
+		if err != nil {
+			return nil, fmt.Errorf("failed to get leaf at index %d: %v", leaf.Index, err)
+		}
+		version, data, err := parseLeaf(leaf.Index, leaf.Data)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse leaf at index %d: %v", leaf.Index, err)
+		}
+		if prev, found := versions[version]; found {
+			return nil, fmt.Errorf("conflicting versions for version %q found!\n%v\n%v", version, prev, data)
```

Comment:
You could have a `var errs []error` and append then, then `return errors.Join(errs...)` so the user sees all colliding versions from a single run?